modules/user/src/main/RankingApi.scala

```diff
@@ -73,6 +73,29 @@ final class RankingApi(
                   )
               .dmap(_.flatten)
 
+  private[user] def topPerfRange(perfId: PerfId, skip: Int, nb: Int): Fu[List[LightPerf]] =
+    lila.rating
+      .PerfType(perfId)
+      .map(_.key)
+      .filter(k => lila.rating.PerfType.isLeaderboardable(PerfType(k)))
+      .so: perfKey =>
+        coll:
+          _.find($doc("perf" -> perfId, "stable" -> true))
+            .sort($doc("rating" -> -1))
+            .skip(skip)
+            .cursor[Ranking]()
+            .list(nb)
+            .flatMap:
+              _.parallel: r =>
+                lightUser(r.user).map2: light =>
+                  LightPerf(
+                    user = light,
+                    perfKey = perfKey,
+                    rating = r.rating,
+                    progress = ~r.prog
+                  )
+            .dmap(_.flatten)
```

Comment:
Looks like 20 lines copy-pasted from `topPerf` above, with the addition of `.skip(skip)`.
Code duplication is usually frown upon for maintenance reasons, here it would be very easy to refactor (`.skip(0)`)